lib/extras/dec/apng.cc

```diff
@@ -405,6 +405,14 @@ Status DecodeImageAPNG(const Span<const uint8_t> bytes,
 
   std::vector<FrameInfo> frames;
 
+  auto before_return = [&](void*) {
+    // It is safe to invoke this method repeatedely; so make sure memory is
+    // released in any case.
+    png_destroy_read_struct(&png_ptr, &info_ptr, 0);
+  };
+  std::unique_ptr<void, decltype(before_return)> scope_guard((void*)1,
+                                                             before_return);
```

Comment:
I'd rather

struct Guard {
 ~Guard() { ... }
} guard;

Reply:
Sure. It seems that there are 2 guards already -> time to extract template for that.